File: BlitzBi/Module/BlitzBiEventHandlerBuilder.m
//
//  BlitzBiEventHandlerBuilder.m
//  BlitzBi
//
//  Created by Akash on 01/10/20.
//  Copyright © 2016 Blitz, Inc. All rights reserved.
//

#import <BlitzBiEventHandlerBuilder.h>

@implementation BlitzBiEventHandlerBuilder
- (void)setParams:(NSNumber*)batchSize
            withUrl: (NSString*)baseUrl
            withAppId:(NSString*)appId
            withAppToken:(NSString*) appToken
            withService:(id <PBlitzDataTransferService>) networkService {
    self->batchSize = batchSize;
    self->baseUrl = baseUrl;
    self->appId = appId;
    self->appToken = appToken;
    self->networkService = networkService;
}

- (BlitzBiEventSendHandler*) build {
    return [[BlitzBiEventSendHandler alloc] init:batchSize :baseUrl :[[BlitzBiEventRepository alloc]init:appId :appToken :networkService]];;
}
@end

File: BlitzBi/Module/BlitzBiEventHandlerBuilder.swift
//
//  BIEventHandlerBuilder.swift
//  Bi
//
//  Created by Saurabh on 20/09/20.
//

import Foundation

public class BlitzBIEventHandlerBuilder {
    private var batchSize:Int!
    private var baseUrl:String!
    private var networkService:PBlitzDataTransferService!
    private var appId: Int!
    private var appToken: String!
    
    public static func setParams(batchSize:Int, baseUrl: String, networkService:PBlitzDataTransferService, appId: Int, appToken: String) -> BlitzBIEventHandlerBuilder {
        let builder = BlitzBIEventHandlerBuilder()
        builder.batchSize = batchSize
        builder.baseUrl = baseUrl
        builder.networkService = networkService
        builder.appId = appId
        builder.appToken = appToken
        return builder
    }
    
    public func build() -> PBlitzBiEventSendHandler {
        return BlitzBiEventSendHandler(batchSize: batchSize, baseUrl: baseUrl, eventRepository: BlitzBIEventRepository(networkService: networkService, appId: appId, appToken: appToken))
    }
}

